postgres/datadog_checks/postgres/connections.py

```diff
@@ -65,9 +107,34 @@ def get_connection(self, dbname, ttl_ms):
                 db.rollback()
 
             deadline = datetime.datetime.now() + datetime.timedelta(milliseconds=ttl_ms)
-            self._conns[dbname] = ConnectionWithTTL(db, deadline)
+            self._conns[dbname] = ConnectionInfo(
+                connection=db,
+                deadline=deadline,
+                active=True,
+                last_accessed=datetime.datetime.now(),
+                thread=threading.current_thread(),
+            )
             return db
 
+    @contextlib.contextmanager
+    def get_connection(self, dbname: str, ttl_ms: int, timeout: int = None):
+        """
+        Grab a connection from the pool if the database is already connected.
+        If max_conns is specified, and the database isn't already connected,
+        make a new connection if the max_conn limit hasn't been reached.
+        Blocks until a connection can be added to the pool,
+        and optionally takes a timeout in seconds.
+        """
+        try:
+            with self._mu:
+                db = None
```

Comment:
What is the purpose of this line?

Reply:
if `_get_connection_raw` raises an exception in `try`, then I need some way to transfer information between the `try` and `finally` block about that having failed, so the `dbname` active flag doesn't get set. without this line `db` never gets instantiated and raises an error when it's referenced in the finally block, so I instantiated it to None. I thought also about instead using a boolean `success` flag that I could set if the try block completed. is there another way to do that?

Reply:
Ok that should work!